phpBB/assets/javascript/phpbb-avatars.js

```diff
@@ -0,0 +1,174 @@
+(function($) {  // Avoid conflicts with other libraries
+
+'use strict';
+
+/**
+ * phpBB Avatars namespace.
+ *
+ * Handles cropping for local file uploads.
+ */
+phpbb.avatars = {
+	cropper: null,
+	image: null,
+
+	buttons: $('#avatar-cropper-buttons'),
+	box: $('#avatar-box'),
+	data: $('#avatar-cropper-data'),
+	input: $('#avatar_upload_file'),
+	driver: $('#avatar_driver'),
+	driverUpload: 'avatar_driver_upload',
+
+	/**
+	 * Initialise avatar cropping.
+	 */
+	init: function() {
+		// If the cropper library is not available
+		if (!$.isFunction($.fn.cropper)) {
+			return;
+		}
+
+		// Correctly position the cropper buttons
+		this.buttons.appendTo(this.box);
+
+		this.image = this.box.children('img');
+
+		this.bindInput();
+		this.bindSelect();
+	},
+
+	/**
+	 * Destroy (undo) any initialisation.
+	 */
+	destroy: function() {
+		this.buttons.find('[data-cropper-action]').off('click.phpbb.avatars')
+		this.image.off('crop.phpbb.avatars');
+
+		this.data.val('');
+		this.buttons.hide();
+
+		if (this.cropper !== null) {
+			this.cropper.destroy();
+		}
+	},
+
+	/**
+	 * Bind a function to the avatar driver <select> element.
+	 *
+	 * If a different driver than the "upload" driver is selected, the cropper is destroyed.
+	 * Otherwise if the "upload" driver is (re-)selected, and it has a value, initialise it.
+	 */
+	bindSelect: function() {
+		this.driver.on('change', function() {
+			if ($(this).val() === phpbb.avatars.driverUpload) {
+				if (phpbb.avatars.input.val() !== '') {
+					phpbb.avatars.input.trigger('change');
+				}
+			} else {
+				phpbb.avatars.destroy();
+			}
+		});
+	},
+
+	/**
+	 * Bind a function to the avatar file upload <input> element.
+	 *
+	 * If a file was chosen and it is a valid image file, the cropper is initialised.
+	 * Otherwise the cropper is destroyed.
+	 */
+	bindInput: function() {
+		this.input.on('change', function() {
+			var fileReader = new FileReader;
```

Comment:
Let's use new ES style syntax, i.e. `let` or `const`

In this case I think you can actually use const